docs/en/guides/pipeline-integrations/development/sgtk-developer-app.md

```diff
@@ -0,0 +1,257 @@
+    description: The template to use when building the path to save the file into
+    allows_empty: False
+```
+Creating a setting for this, means you don't have to hard code the template name in your app code, 
+and [can instead get the value](https://developer.shotgunsoftware.com/tk-core/platform.html?highlight=get_setting#sgtk.platform.Application.get_setting) 
+from the settings defined either by default in the `info.yml` or overridden in the environment YAML file settings.
+
+```python
+template = app.get_setting("save_template")
+```
+This means that you could configure your app to use a different template depending on the environment the app is running in.
+
+You can read more on configuration settings [in the reference documentation](https://developer.shotgunsoftware.com/tk-core/platform.html?highlight=allows_empty#the-configuration-section).
+
+### Frameworks
+
+If you know that you will be using frameworks, you can add them to the manifest file.
+
+For example, if you plan to use the widget and shotgun utils frameworks for your app, add the following to the `info.yml`:
+
+```python
+# the frameworks required to run this app
+frameworks:
+    - {"name": "tk-framework-shotgunutils", "version": "v2.x.x"}
+    - {"name": "tk-framework-qtwidgets", "version": "v1.x.x"}
+```
+
+#### Minimum required framework versions
+If there is a required minimum version for a framework, the minimum_version setting can be used in the `info.yml`:
+
+```python
+# the frameworks required to run this app
+frameworks:
+    - {"name": "tk-framework-shotgunutils", "version": "v2.x.x"}
+    - {"name": "tk-framework-qtwidgets", "version": "v1.x.x", "minimum_version": "v1.5.0"}
+```
+
+The above will ensure that `v1.5.0` of `tk-framework-qtwidgets` is available for the app to use. 
+If it is not, the app will not be loaded at startup time and an error will be printed to the console. 
+When the app is updated using `tank updates`, any configured frameworks not meeting their required minimum versions will be automatically updated along with the app.
+
+For more information about frameworks and how they can be useful, check out the following links:
+
+- [The Qt Widgets Framework](https://developer.shotgunsoftware.com/tk-framework-qtwidgets/)
+- [The Shotgun utils Framework](https://developer.shotgunsoftware.com/tk-framework-shotgunutils/)
+
+### Reloading your changes
+
+If you are testing your app within software such as Maya, then as soon as you have one more dev items in your configuration, 
+Toolkit will automatically add a **Reload and Restart** option to the Shotgun menu.
+![Shotgun menu Reload and Restart option](./images/reload-restart.png)
+
+Clicking this will reload your configuration and code and then restart your engine. 
+This means that you can iterate quickly: Start Maya once, and then each time you make code or configuration changes that you want to try out, simply hit the reload and restart button and your changes will be pulled in. 
```

Comment:
capitalization stuff:
 - I believe "start" should be lower-case
 - "simply hit the Reload and Restart button"